src/Servers/Kestrel/Core/src/Internal/Infrastructure/PipeWriterHelpers/TimingPipeFlusher.cs

```diff
@@ -97,6 +97,10 @@ private async ValueTask<FlushResult> TimeFlushAsyncAwaited(ValueTask<FlushResult
             {
                 outputAborter.Abort(new ConnectionAbortedException(CoreStrings.ConnectionOrStreamAbortedByCancellationToken, ex));
             }
+            catch (ConnectionResetException ex) when (outputAborter is object)
+            {
+                outputAborter.Abort(new ConnectionAbortedException(CoreStrings.ConnectionOrStreamAbortedByTransport, ex));
```

Comment:
Our transports are not supposed to throw on writes when a client disconnected. As noted below, `// A canceled token is the only reason flush should ever throw.`

Reply:
Does that mean the Output transport should only ever be completed with no error, or OperationCanceledException?

https://github.com/dotnet/aspnetcore/blob/5fdf28db6257fb373858eeb0e2b5db85f14aacbf/src/Servers/Kestrel/Transport.Quic/src/Internal/QuicStreamContext.cs#L321-L404

Reply:
I looked at what the socket transport does, and updated QUIC to follow it. Now it is only completed with unexpected exceptions.